### ISSUE TITLE
[Bug] Creating Client Point of Contact

### ISSUE THREAD

**Issue by tevers200** [brainthee/CHAOTICA#45]
When attempting to create a new client point of contact, CHAOTICA dies with a 500 internal server error:

Target URL: `/client/example-client/contact/create/`

Steps to reproduce:
1. Create a new client via the `/client/create/` interface
2. Select the new client from the list of clients at `/clients/`
3. Click the "Contacts" button (`/client/example-client/#contacts`)
4. Click "Add Contact" (`/client/example-client/contact/create/`)
5. 500 error returned

This also prevents other application functionality such as creating jobs from working as intended, as you must select the client contact at the job creation stage:
![image](https://github.com/brainthee/CHAOTICA/assets/4094504/10bbe5cb-52c6-4877-9498-c697582149b9)

I can't find any logs from the application, but if you let me know where these are stored (if they are created), I can provide additional information if it helps.

**Comment by brainthee** [comment]
Managed to replicate it:

```
Internal Server Error: /client/mrdrybones/contact/create/
Traceback (most recent call last):
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/sentry_sdk/integrations/django/views.py", line 84, in sentry_wrapped_callback
    return callback(request, *args, **kwargs)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/guardian/mixins.py", line 210, in dispatch
    response = self.check_permissions(request)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/guardian/mixins.py", line 176, in check_permissions
    obj = self.get_permission_object()
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/guardian/mixins.py", line 166, in get_permission_object
    return (hasattr(self, 'get_object') and self.get_object() or
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/views/generic/detail.py", line 42, in get_object
    queryset = queryset.filter(**{slug_field: slug})
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/query.py", line 1436, in filter
    return self._filter_or_exclude(False, args, kwargs)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/query.py", line 1454, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/query.py", line 1461, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1534, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1565, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1415, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1225, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
  File "/home/adrian/.virtualenvs/chaotica/lib/python3.10/site-packages/django/db/models/sql/query.py", line 1713, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'slug' into field. Choices are: company, company_id, department, email, first_name, id, jobs_contact_for, jobs_poc_for, jobtitle, last_name, mobile, phone, salutation
```

**Comment by brainthee** [comment]
So the issue is related to how Guardian enforces permission. Basically it's expecting a Contact object on a CreateView to check for permissions but one doesn't exist yet. There was also some confusing behaviour because of the naming of the client slug parameter in the URL.

TLDR; it was checking permissions for an object that doesn't exist yet! I've fixed this but it's also shown some deep need for making the client contact UX much better
